src/main/java/com/github/matidominati/medicalclinic/exception/ApiRequestException.java

```diff
@@ -0,0 +1,11 @@
+package com.github.matidominati.medicalclinic.exception;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.springframework.http.HttpStatus;
+@AllArgsConstructor
+@Getter
+public class ApiRequestException extends RuntimeException{
```

Comment:
Po 1 dlaczego request?

Zaprojektowalbym to troche inaczej zrobilbym go jako MedicalException <- i uzywal jako podstawowego wyjatku dla naszego systemu. Co to oznacza?
- taka klasa moglaby byc abstrakcyjna (abstract). -> wtedy poniekad zmusilibysmy ludzi do tworzenia wyjatkow i uzywania bardziej szczegolowych z nich -> dlaczego? 
1. Wiecej one mowia 
2. W handlerze mozemy je np obslugiwac w bardziej customowy sposob.
 
 Zdefiniuj bardziej szczegolowe wyjatki i ich uzywaj. 
 Jesli zdefiniujemy szczegolowe wyjatki np: DataNotFoundException  -> to wtedy zauwazmy ze ten wyjatek bedzie rozszerzal nasz wyjatek ApiRequestException a co za tym idzie bedzie posiadal wszystkie te same pola -> w jaki sposob zaincjalizowac te pola? -> Poprzez konstruktor uzywaja super(args).
 Zastanowmy sie jak moglby wygladac taki konstruktro do klasy 
 DataNotFoundException -> Jaki httpStatus? -> 404 NOT FOUND
  a wiec w konstruktorze dla ApiRequestException (bazowy wyjatek) bedziemy mieli wszystkie argumenty natomiast w szczegolowych wyjatkach jesli nie ma potrzeby to nie definiujemy konstruktorow przyjmujacych wszystkie argumenty tylko te ktore sa wazne dla tego konkretnego wyjatku.
  Czyli dla DataNotFoundException  konstruktor moglby wygladac nastepujaco:
  
DataNotFoundException(String message) {
super (message, HttpStatus.NOT_FOUND);
}